core/src/autogluon/core/models/ensemble/fold_fitting_strategy.py

```diff
@@ -0,0 +1,144 @@
+import copy
+import logging
+import os
+import time
+from abc import abstractmethod
+
+from ...utils.exceptions import TimeLimitExceeded
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractFoldFittingStrategy(object):
+
+    @abstractmethod
+    def schedule_fold_model_fit(self, model_base, fold_ctx, kwargs):
+        """
+        Schedule fold model training. By design this part is supposed to be 'lazy' evaluator no actual training is performed here.
+        Distributed fitters will handle jobs scheduling here.
+        """
+        pass
+
+    @abstractmethod
+    def after_all_folds_scheduled(self):
+        """
+        Method is called when all the folds are scheduled.
+        Local fitters will perform training here.
+        Distributed fitters will handle job handles and results retrieval here.
+        """
+        pass
+
+
+class SequentialLocalFoldFittingStrategy(AbstractFoldFittingStrategy):
+    """
+    This strategy fits the folds locally in a sequence.
+    """
+
+    def __init__(self, bagged_ensemble_model, X, y, sample_weight, time_limit, time_start, models, oof_pred_proba, oof_pred_model_repeats):
+        """
+        Parameters
+        ----------
+        path : str
+            Directory location to store all outputs.
+        params: dict
+            Fitting parameters
+
+        """
+        self.X = X
+        self.y = y
+        self.sample_weight = sample_weight
+        self.time_limit = time_limit
+        self.time_start = time_start
+        self.models = models
+        self.oof_pred_proba = oof_pred_proba
+        self.oof_pred_model_repeats = oof_pred_model_repeats
+        self.bagged_ensemble_model = bagged_ensemble_model
+        self.jobs = []
+
+    def schedule_fold_model_fit(self, model_base, fold_ctx, kwargs):
+        self.jobs.append([model_base, fold_ctx, kwargs])
+
+    def after_all_folds_scheduled(self):
```

Comment:
Perhaps a clearer name?

Reply:
something like run_all()?

Reply:
I don't want to be prescriptive here because 'schedule' means different things for different schedulers. For local ones it's add to queue to be ran later in in this case after_all_folds_scheduled==run_all, but for distributed ones, it will actually send a job to run on remote host immediately and after_all_folds_scheduled==wait_all. So meaning of this method will change, and prescribing what should happen, I refer to when this method will be called - this makes it more readable in the code.